Fix block/update.ts - NotionCLIErrorFactory.apiError() does not exist

@claude Please fix the non-existent factory method in block/update.ts

**Problem:**
Line 188 in `src/commands/block/update.ts` calls `NotionCLIErrorFactory.apiError()` which doesn't exist.

**Compilation error:**
```
src/commands/block/update.ts(188,35): error TS2339: Property 'apiError' does not exist on type 'typeof NotionCLIErrorFactory'.
```

**Available factory methods:**
Looking at `src/errors/enhanced-errors.ts` lines 200-568, the NotionCLIErrorFactory has these methods:
- `tokenMissing()`
- `tokenInvalid()`
- `integrationNotShared(resourceType, resourceId?)`
- `resourceNotFound(resourceType, identifier)`
- `invalidIdFormat(input, resourceType?)`
- `databaseIdConfusion(attemptedId)`
- `workspaceNotSynced(databaseName)`
- `rateLimited(retryAfter?)`
- `invalidJson(jsonString, parseError)`
- `invalidProperty(propertyName, databaseId?)`
- `networkError(originalError)`

But no `apiError()` method.

**Solution:**
Check the context at line 188 in `src/commands/block/update.ts` to understand what error is being created, then either:
1. Use the appropriate existing factory method
2. Or create a new NotionCLIError directly with appropriate code and message

**Tasks:**
1. Read line 188 context in `src/commands/block/update.ts`
2. Determine what error scenario is being handled
3. Replace `NotionCLIErrorFactory.apiError(...)` with either:
   - An appropriate existing factory method, OR
   - A direct `new NotionCLIError()` call with proper code/message
4. Verify the fix compiles

**Files to modify:**
- `src/commands/block/update.ts` (line 188)

**This is the LAST compilation error!**

--- a/src/commands/block/update.ts
+++ b/src/commands/block/update.ts
@@ -10,6 +10,7 @@ import { resolveNotionId } from '../../utils/notion-resolver'
 import { AutomationFlags } from '../../base-flags'
 import {
   NotionCLIError,
+  NotionCLIErrorCode,
   NotionCLIErrorFactory,
   wrapNotionError
 } from '../../errors'
@@ -185,7 +186,7 @@ export default class BlockUpdate extends Command {
         // Ensure we have a full block response
         if (!isFullBlock(blockResponse)) {
           throw new NotionCLIError(
-            NotionCLIErrorFactory.apiError('block', blockId).code,
+            NotionCLIErrorCode.API_ERROR,
             'Received partial block response. Cannot determine block type for color update.',
             [],
             { attemptedId: blockId }

